types/i/index.d.ts

```diff
@@ -0,0 +1,375 @@
+    uncountability(word: string): boolean;
+
+    /**
+     * Returns the plural form of the word in the string.
+     *
+     *     "post".pluralize()             // => "posts"
+     *     "octopus".pluralize()          // => "octopi"
+     *     "sheep".pluralize()            // => "sheep"
+     *     "words".pluralize()            // => "words"
+     *     "CamelOctopus".pluralize()     // => "CamelOctopi"
+     */
+    pluralize(word: string): string;
+
+    /**
+     * The reverse of _pluralize_, returns the singular form of a word in a string.
+     *
+     *     "posts".singularize()            // => "post"
+     *     "octopi".singularize()           // => "octopus"
+     *     "sheep".singularize()            // => "sheep"
+     *     "word".singularize()             // => "word"
+     *     "CamelOctopi".singularize()      // => "CamelOctopus"
+     */
+    singularize(word: string): string;
+
+    /**
+     * Capitalizes the first word and turns underscores into spaces and strips a
+     * trailing "_id", if any. Like _titleize_, this is meant for creating pretty output.
+     *
+     *     "employee_salary".humanize()   // => "Employee salary"
+     *     "author_id".humanize()         // => "Author"
+     */
+    humanize(lower_case_and_underscored_word: string): string;
+
+    /**
+     * Capitalizes all the words and replaces some characters in the string to create
+     * a nicer looking title. _titleize_ is meant for creating pretty output. It is not
+     * used in the Bullet internals.
+     *
+     *
+     *     "man from the boondocks".titleize()   // => "Man From The Boondocks"
+     *     "x-men: the last stand".titleize()    // => "X Men: The Last Stand"
+     */
+    titleize(word: string): string;
+
+    /**
+     * Create the name of a table like Bullet does for models to table names. This method
+     * uses the _pluralize_ method on the last word in the string.
+     *
+     *     "RawScaledScorer".tableize()   // => "raw_scaled_scorers"
+     *    "egg_and_ham".tableize()       // => "egg_and_hams"
+     *     "fancyCategory".tableize()     // => "fancy_categories"
+     */
+    tableize(class_name: string): string;
+
+    /**
+     * Create a class name from a plural table name like Bullet does for table names to models.
+     * Note that this returns a string and not a Class.
+     *
+     *     "egg_and_hams".classify()   // => "EggAndHam"
+     *     "posts".classify()          // => "Post"
+     *
+     * Singular names are not handled correctly:
+     *
+     *     "business".classify()       // => "Busines"
+     */
+    classify(table_name: string): string;
+
+    /**
+     * Gives easy access to add inflections to this class
+     */
+    inflect(fn: (inflections: Inflections) => void): void;
+}
+
+/**
+ * @param attach if set to true all function can be called directly on a string otherwise they will return undefined
+ */
+declare function I(attach?: boolean): Iinterface;
+
+export = I;
+
+declare global {
```

Comment:
These global augmentations only get added when `true` is passed to the `I` function defined here. Given that conditionality, I'm inclined to say  these should be separated into a separate `globals.d.ts` that consumers can load if they want, rather then this package by default polluting the global namespace (since it doesn't pollute by default at runtime!),

Reply:
Right, thanks for the advice, I hope this is the correct way to do it